nestjs/src/courses/team-distribution/team-distribution-student.service.ts

```diff
@@ -41,8 +41,8 @@ export class TeamDistributionStudentService {
     }
   }
 
-  private verifyStudentTeams(students: Student[], teamDistributionId: number) {
-    if (students.find(st => st.teams.find(t => t.teamDistributionId === teamDistributionId))) {
+  private verifyStudentTeams(students: Student[], teamDistributionId: number, teamId?: number) {
+    if (students.find(st => st.teams.find(t => t.teamDistributionId === teamDistributionId && t.id !== teamId))) {
```

Comment:
Once again about abbreviations + I think `some` is more sematic here, we need only flag, not value
```suggestion
    if (
      students.some(student =>
        student.teams.some(team => team.teamDistributionId === teamDistributionId && team.id !== teamId),
      )
    ) {
```

Reply:
fixed